service/worker/deletenamespace/deleteexecutions/activities.go

```diff
@@ -154,3 +164,33 @@ func (a *Activities) DeleteExecutionsActivity(ctx context.Context, params Delete
 	}
 	return result, nil
 }
+
+func (a *Activities) deleteWorkflowExecutionFromVisibility(
+	ctx context.Context,
+	namespaceID namespace.ID,
+	execution *workflowpb.WorkflowExecutionInfo,
+) (successCount int, errorCount int) {
+
+	a.logger.Info("Deleting workflow execution from visibility.", tag.WorkflowNamespaceID(namespaceID.String()), tag.WorkflowID(execution.Execution.GetWorkflowId()), tag.WorkflowRunID(execution.Execution.GetRunId()))
+	_, err := a.historyClient.DeleteWorkflowVisibilityRecord(ctx, &historyservice.DeleteWorkflowVisibilityRecordRequest{
+		NamespaceId:       namespaceID.String(),
+		Execution:         execution.GetExecution(),
+		WorkflowStartTime: execution.GetStartTime(),
+		WorkflowCloseTime: execution.GetCloseTime(),
+	})
+	switch err.(type) {
+	case nil:
+		// Indicates that main and visibility stores were in inconsistent state.
+		a.metricsHandler.Counter(metrics.DeleteExecutionsSuccessCount.GetMetricName()).Record(1)
+		a.logger.Info("Workflow execution deleted from visibility.", tag.WorkflowNamespaceID(namespaceID.String()), tag.WorkflowID(execution.Execution.GetWorkflowId()), tag.WorkflowRunID(execution.Execution.GetRunId()))
+		return 1, 0
+	case *serviceerror.NotFound:
```

Comment:
DeleteWorkflowVisibilityRecord will return not found?

Reply:
I hope no, but there is no clear contract here. No harm to handle it separately.